### PR TITLE
Lightcurve Plotting Refactor and Bug fix

### PR DESCRIPTION
**Pull request by ehsteve**
This introduce refactors out peek from the sub-classes and introduces plot_types so each LC can define any number of plot types. Also cleans up and standardizes plots across all LC sources.

You can preview the changes in the following [ipython notebook](https://github.com/ehsteve/ipython-notebooks/blob/master/Testing%20LC%20plot%20Fixes%200.6.ipynb)
## Needs to be done
- [ ] Address code comments
- [ ] Changelog entry

### CODE REVIEW

File: sunpy/lightcurve/lightcurve.py [hunk 1]
```diff
@@ -210,15 +216,14 @@ def plot(self, axes=None, **plot_args):
 
     def peek(self, **kwargs):
         """Displays the light curve in a new figure"""
-
+        num_plots = len(self._get_plot_types())
```
Comment (ayshih) [inline sunpy/lightcurve/lightcurve.py]: Use your property `plot_types` here

File: sunpy/lightcurve/lightcurve.py [hunk 2]
```diff
@@ -210,15 +216,14 @@ def plot(self, axes=None, **plot_args):
...
         figure = plt.figure()
 
-        self.plot(**kwargs)
-
+        for plot_type, plot_num in zip(self.plot_types, np.arange(0, num_plots)):
```
Comment (ayshih) [inline sunpy/lightcurve/lightcurve.py]: Why not just `enumerate`?

**Comment by ayshih** [comment]
Why do you use the approach of `self._get_plot_types()`?  It seems like it just returns a static list in all of the derived classes, so wouldn't it be more straightforward to just have a class attribute?  Plus, that makes it easier if a user wants to customize the plot types for a given object (e.g., to remove one from showing up in a `peek()` call).

File: sunpy/lightcurve/sources/lyra.py [hunk 3]
```diff
@@ -38,49 +38,44 @@ class LYRALightCurve(LightCurve):
     | http://proba2.sidc.be/data/LYRA
     """
 
-    def peek(self, names=3, **kwargs):
-        """Plots the LYRA data
+    def plot(self, title="LYRA", axes=None, plot_type='channel 1', **plot_args):
+        """Plots the LYRA data. Available plot types are
```
Comment (ayshih) [inline sunpy/lightcurve/sources/lyra.py]: Incomplete docstring

File: sunpy/lightcurve/sources/norh.py [hunk 4]
```diff
@@ -38,20 +38,29 @@ class NoRHLightCurve(LightCurve):
...
+        axes.yaxis.grid(True, 'major')
+        axes.xaxis.grid(True, 'major')
+        axes.legend(bbox_to_anchor=(1.02, 1), loc=2, borderaxespad=0.)
+        plt.gcf().autofmt_xdate()
 
+        return axes
+
+    @classmethod
+    def _get_plot_types(cls):
+        return ['norh']
```
Comment (ayshih) [inline sunpy/lightcurve/sources/norh.py]: Add another newline

**Comment by ayshih** [comment]
Can you make the argument order consistent across the different `plot()` calls?  It's weird.

File: sunpy/lightcurve/sources/eve.py [hunk 5]
```diff
@@ -34,28 +37,54 @@ class EVELightCurve(LightCurve):
...
+            axes.set_title(title)
+            axes.yaxis.grid(True, 'major')
+            axes.xaxis.grid(True, 'major')
+            axes.legend(bbox_to_anchor=(1.02, 1), loc=2, borderaxespad=0.)
+            plt.gcf().autofmt_xdate()
+
+            return axes
+
+    @classmethod
+        def _get_plot_types(cls):
```
Comment (ayshih) [inline sunpy/lightcurve/sources/eve.py]: Fix indentation, can't even import!

File: sunpy/lightcurve/sources/goes.py [hunk 6]
```diff
@@ -34,23 +34,26 @@ class GOESLightCurve(LightCurve):
     | http://umbra.nascom.nasa.gov/goes/fits/
     """
 
-    def peek(self, title="GOES Xray Flux"):
-        """Plots GOES light curve is the usual manner"""
-        figure = plt.figure()
-        axes = plt.gca()
+    @classmethod
+        def _get_plot_types(cls):
```
Comment (ayshih) [inline sunpy/lightcurve/sources/goes.py]: Fix indentation, can't even import!

**Comment by ehsteve** [comment]
@ayshih The idea of using a class property is a good one. The problem is that we current do not instantiate these subclasses with __init__ where I could update that property. This could be fixed in the lightcurve factory.

**Comment by Cadair** [comment]
@ehsteve just make it a property using the `@property` decorator?

**Comment by ehsteve** [comment]
@Cadair But how do I set it in the subclass?

**Comment by Cadair** [comment]
@ehsteve redefine the property.

**Comment by Cadair** [comment]
@ehsteve Should I close this until you update it?

**Comment by ehsteve** [comment]
this is very stale but still important. i will work on updating.

**Comment by ehsteve** [comment]
@Cadair this includes a lot of good work but I'm not sure i'll be able to get it to competition. Can you take a look at it please?

**Comment by Cadair** [comment]
@ehsteve I will try, I am not sure I am going to be able to finish it before 0.7 though.

**Comment by ehsteve** [comment]
@DanRyanIrish

**Comment by Cadair** [comment]
@ehsteve I am going to close this as it needs reworking for timeseries.

**Pull request closed without merging**